LjinuxRoot/etc/jpkg/tools/generatelist.py

```diff
@@ -0,0 +1,41 @@
+"""
+JPKG automatic list loader
+
+Returns in user_vars["return"] a dict with every package name and its deps / conflicts
+{package: [list(deps), list(conflicts)]}
+"""
+
+stdout.write("Generating package list.. 0%")
+listing = listdir("/LjinuxRoot/etc/jpkg/installed/")
+pkc = len(listing)  # package count
+pkl = 0  # packages loaded
+cc = 2  # currently displays characters
+
+installed = dict()  # All packages
+dependencies = set()  # All dependencies
+conflicts = set()  # All conflicts
+
+for i in listing:
+    name = i[:-5]  # remove ".json"
+    with ljinux.based.fn.fopen("/etc/jpkg/installed/" + i) as conf_f:
+        conf = json.load(conf_f)
+        installed.update(
+            {name: [conf["version"], conf["dependencies"], conf["conflicts"]]}
+        )
+        for j in conf["dependencies"]:
```

Comment:
Be more expressive, stop spamming i and j k l ,  etc

Reply:
I really should stop doing that